File: problems/119-Pascal's-Triangle-II/solution.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> result = new LinkedList<>();
        result.add(1);

        while (rowIndex > 0) {
            List<Integer> temp = new LinkedList<>();

            temp.add(1);
            for (int i = 1; i < result.size(); i++) {
                temp.add(result.get(i - 1) + result.get(i));
            }
            temp.add(1);

            result = temp;

            rowIndex--;
        }

        return result;
    }
}

File: problems/119-Pascal's-Triangle-II/solution.go
func getRow(rowIndex int) []int {
	rowIndex++
	result, temp := make([]int, rowIndex), make([]int, rowIndex)

	for i := 0; i < rowIndex; i++ {
		temp[0] = 1
		temp[i] = 1

		if i == 1 {
			copy(result, temp)
		} else {
			for j := 1; j < i; j++ {
				temp[j] = result[j-1] + result[j]
			}
			copy(result, temp)
		}
	}

	return result
}

